dotnet-estimation/dotnet-estimation/Templates/WebAPI/Devon4Net.Application.WebAPI.Implementation/Business/SessionManagement/Service/SessionService.cs

```diff
@@ -26,6 +26,21 @@ public SessionService(ILiteDbRepository<Session> SessionRepository)
         {
             _sessionRepository = SessionRepository;
         }
+
+        public void validateSession(Session session, long sessionId)
+        {
+
+            if (session == null)
+            {
+                throw new NotFoundException(sessionId);
+            }
+
+            if (!session.IsValid())
```

Comment:
I do not like this pattern. I still think we should rather think about introducing an ErrorOr or ResultType. Generally just because an action cannot be performed it should not cause the program to abort or throw an exception.

A result type would also not require to try catch and only the database operations would be needed to be caught.

Reply:
I've implemented the ErrorOr package and changed the service and controller accordingly